data/src/main/java/org/apache/iceberg/data/BaseFileWriterFactory.java

```diff
@@ -261,6 +264,8 @@ public PositionDeleteWriter<T> newPositionDeleteWriter(
                   .withSpec(spec)
                   .withPartition(partition)
                   .withKeyMetadata(keyMetadata)
+                  .withFileEncryptionKey(keyMetadata.encryptionKey())
```

Comment:
Same here. Why doesn't this use the raw output file?

Also, should we have a `Parquet.writeDeletes(EncryptingOutputFile)`? It would work like this:
```java
  public static DeleteWriteBuilder writeDeletes(EncryptedOutputFile file) {
    if (file.keyMetadata() instanceof KeyMetadata) {
      KeyMetadata standardKeyMetadata = (KeyMetadata) file.keyMetadata();
      return writeDeletes(file.rawOutputFile())
          .keyMetadata(standardKeyMetadata.encryptionKey())
          .withAADPrefix(standardKeyMetadata.aadPrefix());
    } else {
      return writeDeletes(file.encryptingOutputFile());
    }
  }
```